plugins/c9.vfs.client/vfs_client.js

```diff
@@ -193,12 +193,12 @@ define(function(require, exports, module) {
                 extraPaths = path;
                 path = path[0];
                 extraPaths = "," + extraPaths.map(function(p) {
-                    return p[0] == path[0] && p != path ? escape(p) : "";
+                    return p[0] == path[0] && p != path ? encodeURI(p) : "";
```

Comment:
@nightwing If the filename parameter contained a `+`, that was being converted to a space in the downloaded filename, because `escape` leaves `+` alone and `+` encodes a space in URL parameters (but not URL paths). Note that filenames in the path portion of the URL were not affected (i.e., downloading one file containing a `+` works ok, but downloading a folder containing a `+`, or several files of which the first contains a `+`, would give the wrong filename).

I left `vfsUrl` alone because changing it would risk impacting many other VFS calls that I have little knowledge about and cannot test effectively.